src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -94,40 +94,37 @@ public Person toModelType() throws IllegalValueException {
         }
         final Phone modelPhone = new Phone(phone);
 
-        if (email == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Email.class.getSimpleName()));
+        Email modelEmail = null;
+        if (email != null) {
+            if (!Email.isValidEmail(email)) {
+                throw new IllegalValueException(Email.MESSAGE_CONSTRAINTS);
+            }
+            modelEmail = new Email(email);
         }
-        if (!Email.isValidEmail(email)) {
-            throw new IllegalValueException(Email.MESSAGE_CONSTRAINTS);
-        }
-        final Email modelEmail = new Email(email);
```

Comment:
May I know why this part is changed as such? What's the problem of the original code?

Similar to other fields as well.

Reply:
When saving files, some attributes might be null. This allows attributes with no arguments for email n other optional ones from the json file